integration/spark/src/main/common/java/io/openlineage/spark/agent/util/PlanUtils.java

```diff
@@ -139,6 +141,39 @@ public String getName() {
     };
   }
 
+  public static Path getPath(URI location, String qualifiedName, String authority) {
+    Path path;
+    try {
+      path = new Path(location);
+      if (location.getScheme() == null) {
+        path = new Path("file", null, location.toString());
+      }
+    } catch (Exception e) { // Java does not recognize scala exception
+      if (e instanceof AnalysisException) {
+        path = new Path("hive", authority, qualifiedName);
+      }
+      throw e;
+    }
+    return path;
+  }
+
+  /**
+   * Construct a dataset {@link OpenLineage.Dataset} given a {@link
+   * org.apache.spark.sql.catalyst.catalog.CatalogTable} If table does not contain location, use
+   * passed authority as a backup.
+   *
+   * @param table
+   * @return
+   */
+  public static OpenLineage.Dataset getDataset(CatalogTable table, String authority) {
+    Path path = getPath(table.location(), table.qualifiedName(), authority);
+    return getDataset(path.toUri(), table.schema());
+  }
```

Comment:
I noticed this does funky stuff with Hive tables. The table location is typically the path to the data in storage- e.g., `s3://path/to/the/data`. This path is unique to the table and is, oftentimes, enough to identify the data. With this Dataset construction, we'll end up with `namespace=s3://path/to/the/data` and `name=the_data`, which I don't think we want. I think we probably want either the table location _or_ the table name, but not both. Though attaching the table name as a facet on the dataset is definitely a useful approach.